libfwupdplugin/fu-firmware.c

```diff
@@ -912,7 +912,15 @@ fu_firmware_parse_full(FuFirmware *self,
 		return FALSE;
 
 	/* always set by default */
-	fu_firmware_set_bytes(self, fw);
+	if (offset == 0x0) {
+		fu_firmware_set_bytes(self, fw);
+	} else {
+		g_autoptr(GBytes) fw_offset = NULL;
+		fw_offset = fu_bytes_new_offset(fw, offset, g_bytes_get_size(fw) - offset, error);
+		if (fw_offset == NULL)
+			return FALSE;
+		fu_firmware_set_bytes(self, fw_offset);
+	}
```

Comment:
It has conflict with `fu_firmware_set_offset(firmware_sub, offset);`. By this https://github.com/fwupd/fwupd/blob/d7596d3ccec2b0faf701f1849754e0e6a5700c61/plugins/genesys/fu-genesys-usbhub-device.c#L2028-L2031 Which one is better, replace `g_bytes_get_region` or remove the `fu_firmware_set_offset(firmware_sub, offset);` ?

Reply:
ohh, i missed that on review -- we shouldn't use g_bytes_get_region as that was introduced in 2.70 and we only depend on GLib 2.68. The fu_bytes_new_offset() change was a valid fix -- it was an oversight as firmware shouldn't have to keep track of the [global] offset in the bytes buffer. Give me a few mins.

Reply:
Okay, fixed; can you take a look again please.